src/Features/Core/Portable/SpellCheck/AbstractSpellCheckSpanService.cs

```diff
@@ -0,0 +1,173 @@
+                switch (classification)
+                {
+                    case ClassificationTypeNames.ClassName:
+                    case ClassificationTypeNames.RecordClassName:
+                    case ClassificationTypeNames.DelegateName:
+                    case ClassificationTypeNames.EnumName:
+                    case ClassificationTypeNames.InterfaceName:
+                    case ClassificationTypeNames.ModuleName:
+                    case ClassificationTypeNames.StructName:
+                    case ClassificationTypeNames.RecordStructName:
+                    case ClassificationTypeNames.TypeParameterName:
+                    case ClassificationTypeNames.FieldName:
+                    case ClassificationTypeNames.EnumMemberName:
+                    case ClassificationTypeNames.ConstantName:
+                    case ClassificationTypeNames.LocalName:
+                    case ClassificationTypeNames.ParameterName:
+                    case ClassificationTypeNames.MethodName:
+                    case ClassificationTypeNames.ExtensionMethodName:
+                    case ClassificationTypeNames.PropertyName:
+                    case ClassificationTypeNames.EventName:
+                    case ClassificationTypeNames.NamespaceName:
+                    case ClassificationTypeNames.LabelName:
+                        break;
+                    default:
+                        return;
+                }
+
+                AddSpan(new SpellCheckSpan(token.Span, SpellCheckKind.Identifier));
+            }
+
+            private void ProcessTriviaList(SyntaxTriviaList triviaList, CancellationToken cancellationToken)
+            {
+                foreach (var trivia in triviaList)
+                    ProcessTrivia(trivia, cancellationToken);
+            }
+
+            private void ProcessTrivia(SyntaxTrivia trivia, CancellationToken cancellationToken)
+            {
+                if (_syntaxFacts.IsRegularComment(trivia))
+                {
+                    AddSpan(new SpellCheckSpan(trivia.Span, SpellCheckKind.Comment));
+                }
+                else if (_syntaxFacts.IsDocumentationComment(trivia))
+                {
+                    ProcessDocComment(trivia.GetStructure()!, cancellationToken);
+                }
+            }
+
+            private void ProcessDocComment(SyntaxNode node, CancellationToken cancellationToken)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                foreach (var child in node.ChildNodesAndTokens())
+                {
+                    if (child.IsNode)
+                    {
+                        ProcessDocComment(child.AsNode()!, cancellationToken);
+                    }
+                    else
+                    {
+                        var token = child.AsToken();
+                        if (token.RawKind == _syntaxFacts.SyntaxKinds.XmlTextLiteralToken)
```

Comment:
for things like typenames in cref, are those already classified as the appropriate `ClassificationTypeNames` (e.g. a cref to a class is classified as `ClassificationTypeNames.ClassName`?

Reply:
so spell checking cares about three things.

1. the identifier where a symbol is *declared*, so you can spell check it's name when creating it.
2. the contents of a string.
3. the non-structure contents of a comment.  

 So, `for things like typenames in cref`, that would count as the structure-contents of a comment, and we won't spell check it.